### PR TITLE
LPS-191300 [CP 2.0] Designate contacts for high priority communications

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[ LPS-191300](https://liferay.atlassian.net/browse/LPS-191300)

### CODE REVIEW

**Comment by Aline160** [comment]
Just starting reviewing :)

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.scss
Comment (pedroliveirav) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.scss:1]: Create a css file in the layout folder and import globally.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx [hunk 1]
```diff
@@ -34,10 +34,60 @@ const SetupLiferayExperienceCloudPage = ({
...
+	};
+
+	const handleNextStep = () => {
+		setStep(step + 1);
+	};
+
+	const handleButtonClick = () => {
+		// eslint-disable-next-line no-unused-expressions
+		step === 1 ? handleOnLeftButtonClick() : handlePreviousStep();
+	};
```
Comment (pedroliveirav) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx:58]: Try to use an 'enum' or a object to identify each step here and in the other steps for better organization if possible.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx [hunk 2]
```diff
@@ -85,116 +146,119 @@ const SetupLiferayExperienceCloudPage = ({
...
+							<hr />
+						</>
+					)}
+				/>
+			)}
+
+			{step === 2 && (
+				<div>
+					<SetupHighPriorityContactForm
+						addContactList={addHighPriorityContacts}
```
Comment (pedroliveirav) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx:238]: Try to use an 'enum' or a object to identify each step here and in the other steps for better organization if possible.

**Review by pedroliveirav — CHANGES_REQUESTED** [review]
I made some suggestions! 🔨🔨

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.scss
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.scss:1]: Created! Thanks :man_astronaut:

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.scss [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	.btn-outline-borderless
+	.btn-outline-secondary
+	:hover {
+	color: #6c6c76;
+}
+
+.input-group-item .input-group-item-shrink :hover {
+	color: #6c6c76;
+	border-color: #6c6c76;
+}
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.scss]: Thanks!

**Comment by Aline160** [comment]
Just starting reviewing :)

**Comment by Aline160** [comment]
Just starting reviewing :)

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/formik-fields/MultiSelect/index.jsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+	const formik = useFormikContext();
+	if (props.required) {
+		validations = validations
+			? [...validations, () => required(values.length)]
+			: [() => required(values.length)];
+	}
+	const handleChange = (item) => {
+		return onChanges(item);
+	};
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/formik-fields/MultiSelect/index.jsx:38]: No need

```suggestion
	
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/formik-fields/MultiSelect/index.jsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+							<ClayIcon symbol="asterisk" />
+						</span>
+					)}
+				</label>
+
+				<ClayMultiSelect
+					{...field}
+					{...props}
+					items={items}
+					onChange={() => handleChange(window.event?.target?.value)}
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/formik-fields/MultiSelect/index.jsx:87]: Never get the event from window, bad practice, use the onChange event instead

```suggestion
					onChange={(event) => onChange(event?.target?.value)}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js [hunk 6]
```diff
@@ -5,11 +5,12 @@
 
 import {Liferay} from '.';
 
-const HEADLESS_BASE_URL = `${window.location.origin}/o/headless-delivery/v1.0`;
+const HEADLESS_BASE_URL_HEADLESS = `${window.location.origin}/o/headless-delivery/v1.0`;
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js:8]:
```suggestion
const HEADLESS_DELIVERY_BASE_URL_ = `${window.location.origin}/o/headless-delivery/v1.0`;
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js [hunk 7]
```diff
@@ -5,11 +5,12 @@
...
+const HEADLESS_BASE_URL = `${window.location.origin}/o/`;
 
 const fetchHeadless = async ({resolveAsJson = true, url}) => {
 	// eslint-disable-next-line @liferay/portal/no-global-fetch
-	const response = await fetch(`${HEADLESS_BASE_URL}${url}`, {
+	const response = await fetch(`${HEADLESS_BASE_URL_HEADLESS}${url}`, {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js:13]:
```suggestion
	const response = await fetch(`${HEADLESS_DELIVERY_BASE_URL}${url}`, {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js [hunk 8]
```diff
@@ -24,4 +25,20 @@ const fetchHeadless = async ({resolveAsJson = true, url}) => {
 	return response;
 };
 
-export {fetchHeadless};
+const getHighPriorityContacts = async (filter) => {
+	// eslint-disable-next-line @liferay/portal/no-global-fetch
+	const response = await fetch(
+		`${HEADLESS_BASE_URL}${`c/highprioritycontactses/?nestedFields='user'&filter=${filter}`}`,
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/api.js:31]:
```suggestion
		`${HEADLESS_BASE_URL}${`c/highprioritycontactses/?nestedFields=user&filter=${filter}`}`,
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/hooks/useSubmitLXCEnvironment.js [hunk 9]
```diff
@@ -115,6 +119,43 @@ export default function useSubmitLXCEnvironment(
...
+							},
+							mutation: deleteHighPriorityContacts,
+							variables: {
+								highPriorityContactsId: item.objectId,
+							},
+						});
+					})
+				);
+
+				await Promise.all(
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/hooks/useSubmitLXCEnvironment.js:137]:
```suggestion
				await Promise.allSettled(
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
@eduardo-diniz thank you for the effort on that, I left suggestions for you

Important: 

Avoid using `List` as variable name, Ideally we just need to pluralize the word to intent it's a list.

**Comment by Aline160** [comment]
Just starting reviewing :)

**Comment by Aline160** [comment]
Just starting reviewing :)

**Comment by Aline160** [comment]
Just starting reviewing :)

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by styller1** [comment]
ci:stop

**Comment by Aline160** [comment]
Just starting reviewing :)

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by eduardo-diniz** [comment]
ci:stop

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Pull request closed without merging**